pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -172,60 +172,95 @@ func (m *ManagerImpl) GetNUMAHints(resource string, amount int) numamanager.Numa
             	Affinity:       false,
         	}
 	}
-    	glog.Infof("Available devices for resource: %v", available)
-    	//Figure this out programatically 
-    	const numSockets int = 2
-    	duplicates := map[int]bool{}
-    	socketsAvail := map[int]int{}
-    	defaultMask := [numSockets]int{}
-    	for availID := range available {
-        	for _, device := range devices[resource] {
-            		glog.Infof("AvailID: %v DeviceID: %v", availID, device)
-            		if availID == device.ID {
-                		socket := int(device.Socket)
-                		socketsAvail[socket]++
-                		if socketsAvail[socket] >= amount {
-                    			glog.Infof("Socket: %v", socket)
-                    			//Efficiency?
-                    			deviceMask := ""
-                    			socketMask := [numSockets]int{0,0}
-                    			socketMask[socket] = 1
-                    			for _, bit := range socketMask {
-                        			deviceMask += strconv.Itoa(bit)
-                    			}
-                    			glog.Infof("DeviceMask: %v", deviceMask)
-                    			deviceMaskInt, err := strconv.Atoi(deviceMask)
-                    			if err != nil {
-                        			glog.Errorf("Err: Cannot convert deviceMask to int. %v")
-                        			break
-                    			}
-                    			if !duplicates[deviceMaskInt] {
-                        			duplicates[deviceMaskInt] = true
-                        			nm = append(nm, int64(deviceMaskInt))
-                    			}
-                	        }    
-                		//WIP for preferred     
-                		defaultMask[socket] = 1             
-                		break
+    	glog.Infof("[device-manager] Available devices for resource: %v", available)	
+	duplicate_frequency := make(map[int64]int64)
+	var socketValues []int64	
+ 	for availID := range available {
+		for _, device := range devices[resource] {
+			glog.Infof("[device-manager] AvailID: %v DeviceID: %v", availID, device)
+			if availID == device.ID {
+				socket := int64(device.Socket)
+				socketValues = append(socketValues, socket)	//slice of sockets with resource
+				if duplicate_frequency[socket] <= 1 {
```

Comment:
Should this be >= 1? 
if duplicate_frequncy[socket] = nil, want to initalize it as 1 else if its anything greater than or equal than one we want to increment